students/JasneetChandok/Lesson04/mailroom2.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3.6
+
+'''Lesson 3 - Mailroom Part 1'''
+
+
+# Importing Systems Module
+import sys
+
+# Starter list of existing donors
+donors = {'jasneet paramjit': ("jasneet paramjit", [50, 200, 500]), 
+'simran kaur': ("simran kaur", [100, 200, 500]), 
+'manikaran chandok':("manikaran chandok", [100, 200, 500, 680]),
+'manikaran chandok': ("manveen chandok", [10, 200, 65]), 
+'sanjam patwalia': ("sanjam patwalia", [200, 50])} 
+
+
+def find_donor(name):
+    key = name.strip().lower()
+    return donors.get(key)
+
+
+def add_donor(name):
+    name = name.strip()
+    donor = (name, [])
+    donors[name.lower()] = donor
+    return donor
+
+
+# Function for storing and thanking a new donor
+def new_donor():
+    new_donor_name = input("Enter new donors 'Full Name'>>>")
+    new_donor_donation = int(input("\nEnter the 'Amount' donated>>>"))
+    donor = find_donor(new_donor_name)
+    if donor is None:
+        donor = add_donor(new_donor_name)
+    donor[1].append(new_donor_donation)
+    print("\n****New donor information added to list****\n")
+    print(
+        '''
+          Dear {}
+
+          Thank you for your very kind donation of ${:.2f}.
+          It will be put to very good use.
+
+                         Sincerely,
+                            -The Team
+          '''.format(new_donor_name, new_donor_donation))
+
+
+# Function for updating record and thanking existing donor
+def existing_donor():
+    print("List of Existing Donors:\n")
+    print(donors.keys(), '\n')
+    donor_selection = input("Enter donor full name to store the new donation value>>>")
+    donor_donation = int(input("\nEnter the 'Amount' donated>>>"))
+    donor = find_donor(donor_selection)
+    donor[1].append(donor_donation)
+    print("\n****Donation amount has been recorded****\n")
+    print(
+        '''
+              Dear {}
+
+              Thank you for your very kind donation of ${:.2f}.
+              It will be put to very good use.
+
+                             Sincerely,
+                                -The Team
+        '''.format(donor_selection, donor_donation))
+
+
+# Function for sending a thanking you note
+def thankyou_email():
+    donor_category = input("""Specify if new or pick existing donor from the list
+        New Donor: Enter 'new'
+        Existing Donor: Enter 'list'
+        >>>""")
+    donor_category = donor_category.strip().lower()
+    print("\nYou selected:", donor_category, "\n")
+    if donor_category == "new":

```

Comment:
You actually don't want to *force* them into using an existing user if they type list. An easy way to remedy this is to take the logic of the "show list" out of the existing_donor. Show the list, then do the pick depending on the donor name being in the dict.